File: web/src/krate/kspace/dto.ts
import type { KSpace } from "./po";
import type { Varchar } from "@/lib/types";

export type KSpaceListRsp = {
  kspaces: KSpace[];
};

export type KSpaceCommitReq = {
  kspace: KSpace;
};

export type KSpaceCommitRsp = object;

export type KSpaceListReq = object;

export type KSpaceArchiveReq = {
  kspace_name: Varchar<500>;
};
export type KSpaceArchiveRsp = object;

File: lib/backend/src/krate/kspace/dto.rs
use chin_sql::str_type::Varchar;
use serde::{Deserialize, Serialize};

use super::KSpace;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KSpaceListReq {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KSpaceListRsp {
    pub kspaces: Vec<KSpace>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KSpaceCommitReq {
    pub kspace: KSpace,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KSpaceCommitRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KSpaceArchiveReq {
    pub kspace_name: Varchar<500>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KSpaceArchiveRsp {}
